Fix #465 for isapprox with complex arrays

JeffFessler opened the pull request:
Fix #465
The substantive change was `real{T1}` on line 300.

The rest of it is just white space to make it easier for me to sort out why it worked for scalars and not arrays.
The scalar case on line 282 has `real(T)` in it, so now it is matching between scalar and array.

This seems like a bug fix so I put in v1.9.1, but if you prefer v1.10.0 because it now it is new support for complex arrays that's fine too.  Up to you...

sostock commented on src/quantities.jl:
```diff
@@ -275,18 +275,31 @@ for (i,j) in zip((:<, :isless), (:_lt, :_isless))
+    atol = zero(Quantity{real(T),D,U}),
+    kwargs...,
+) where {T,D,U}
+    return isapprox(x.val, y.val; atol=uconvert(unit(y), atol).val, kwargs...)
```
```suggestion
    return isapprox(x.val, y.val; atol=ustrip(unit(y), atol), kwargs...)
```

JeffFessler replied:
I'll commit this suggestion, but I should mention that the `uconvert` version was already in there.

giordano commented on test/runtests.jl:
```diff
@@ -741,6 +741,10 @@ end
 
+        # Issue 465:
+        z = (1+im) * ones(2,3) * 1m
+        @test !isapprox(z, 2z)
```
Maybe also add a positive test like `isapprox(z, z)`?

JeffFessler replied:
Done

JeffFessler commented:
bump - I think I have addressed all the suggestions last summer.  Then I forgot about it, but I just hit this same issue again and "rediscovered" this PR.
Hopefully the only remaining question is whether to call it v0.11.0 or v0.10.1 (I chose the latter).

sostock commented on test/runtests.jl:
```diff
@@ -760,6 +760,11 @@ end
         @test !isapprox(1.0u"m", 1.1u"m"; rtol=0.05)
 
+        # Issue 465:
+        z = (1+im) * ones(2,3) * 1m
```
```suggestion
        z = fill((1+im)m, 2, 3)
```
I think this would be clearer than the multiplication chain.

JeffFessler replied:
agreed!

sostock reviewed (approved):
LGTM. We should resolve some more issues with `isapprox`(#310, ~~support for the `nans` keyword~~), but that can happen in a future PR.
